repository-gcs/src/main/java/io.aiven.elasticsearch.repositories.gcs/GcsStorageSettings.java

```diff
@@ -89,13 +107,18 @@ public static GcsStorageSettings create(final Settings settings) throws IOExcept
         checkSettings(CREDENTIALS_FILE_SETTING, settings);
         checkSettings(PUBLIC_KEY_FILE, settings);
         checkSettings(PRIVATE_KEY_FILE, settings);
+        if (PROXY_PORT.exists(settings) && PROXY_PORT.get(settings) < 0) {
+            throw new IllegalArgumentException("Settings with name " + PROXY_PORT.getKey() + " must be greater than 0");
+        }
         return new GcsStorageSettings(
```

Comment:
One thing a bit unclear to me is why PROXY_PORT is not a part of a GcsStorageSetting object?

Reply:
it is or I miss something